modules/context/context.go

```diff
@@ -257,6 +257,11 @@ func Contexter() macaron.Handler {
 				branchName = repo.DefaultBranch
 			}
 			prefix := setting.AppURL + path.Join(url.PathEscape(ownerName), url.PathEscape(repoName), "src", "branch", util.PathEscapeSegments(branchName))
+
+			insecure := ""
+			if setting.Protocol == setting.HTTP {
```

Comment:
As @lafriks said, gitea maybe behind a proxy.

Reply:
OK, Although setting.Domain is updated with the correct host, I think setting.Protocol can't necessary be trusted to be updated - therefore I've switched to parsing the AppURL.